Source/BugsnagClient.m

```diff
@@ -774,51 +829,46 @@ - (void)batteryChanged:(NSNotification *)notif {
                  toTabWithName:BSGKeyDeviceState];
 }
 
-- (void)orientationChanged:(NSNotification *)notif {
-    NSString *orientation;
-    UIDeviceOrientation deviceOrientation =
-        [UIDevice currentDevice].orientation;
-
-    switch (deviceOrientation) {
-    case UIDeviceOrientationPortraitUpsideDown:
-        orientation = @"portraitupsidedown";
-        break;
-    case UIDeviceOrientationPortrait:
-        orientation = @"portrait";
-        break;
-    case UIDeviceOrientationLandscapeRight:
-        orientation = @"landscaperight";
-        break;
-    case UIDeviceOrientationLandscapeLeft:
-        orientation = @"landscapeleft";
-        break;
-    case UIDeviceOrientationFaceUp:
-        orientation = @"faceup";
-        break;
-    case UIDeviceOrientationFaceDown:
-        orientation = @"facedown";
-        break;
-    default:
-        return; // always ignore unknown breadcrumbs
+/**
+ * Called when an orientation change notification is received to record an
+ * equivalent breadcrumb.
+ *
+ * @param notification The orientation-change notification
+ */
+- (void)orientationChanged:(NSNotification *)notification {
+    UIDeviceOrientation currentDeviceOrientation = [UIDevice currentDevice].orientation;
+    NSString *orientation = BSGOrientationNameFromEnum(currentDeviceOrientation);
+    
+    // Short-circuit the exit if we don't have enough info to record a full breadcrumb
+    // or the orientation hasn't changed (false positive).
+    if (!orientation) {
+        return;
+    }
+    else if (!_lastOrientation || [orientation isEqualToString:_lastOrientation]) {
+        _lastOrientation = orientation;
+        return;
     }
 
-    NSDictionary *lastBreadcrumb =
-        [[self.configuration.breadcrumbs arrayValue] lastObject];
-    NSString *orientationNotifName =
-        BSGBreadcrumbNameForNotificationName(notif.name);
+    NSDictionary *lastBreadcrumb = [[self.configuration.breadcrumbs arrayValue] lastObject];
+    NSString *orientationNotifName = BSGBreadcrumbNameForNotificationName(notification.name);
 
-    if (lastBreadcrumb &&
-        [orientationNotifName isEqualToString:lastBreadcrumb[BSGKeyName]]) {
+    if (lastBreadcrumb && [orientationNotifName isEqualToString:lastBreadcrumb[BSGKeyName]])
+    {
         NSDictionary *metadata = lastBreadcrumb[BSGKeyMetadata];
 
         if ([orientation isEqualToString:metadata[BSGKeyOrientation]]) {
             return; // ignore duplicate orientation event
         }
     }
-
-    [[self state] addAttribute:BSGKeyOrientation
-                     withValue:orientation
+    
+    // Record the breadcrumb
+    [[self state] addAttribute:BSGKeyOrientationChange
+                     withValue:@{@"from" : _lastOrientation,
+                                 @"to" : orientation}
```

Comment:
This is the call to add metadata - think won't be adding it to the breadcrumb (as far as I can see), this seems to be done through `sendBreadcrumbForNotification`.

Reply:
Doh.  Resolved after discussion.